### ISSUE TITLE
Documentation needed to explain why program opens with blank screen

### ISSUE THREAD

**Issue by patowen** [patowen/hyperworld#1]
Reported by powerapproach:

"I did the full static build - exe created without issue.  Program screen is blank, then bails.  Running under Windows 10.  Any clues?  Is there a forum?  I have run the exe out of the build directory.  Do I need to do some additional make or install process?"

**Comment by patowen** [comment]
TL;DR: Can you try opening a command prompt window and launching the program from there?


Long version:

In the same folder as the executable, there should be a folder called "shaders" and a folder called "textures". Double-clicking the executable should open a command prompt window followed by the actual Hyperworld window, which after a bit of loading should be all black. Pressing a number key such as 1, 4, or 5 on the main keyboard (not the numpad) should spawn an object to look at. Normally, if there's an error, it should appear on the command prompt window, and it will remain open until you press enter.

If the command prompt window doesn't show anything or closes by itself, that suggests there's a bug in the application itself, although I can't tell what it is at the moment. Launching it from a command prompt window will allow that window to remain open even after the program crashes and may reveal additional clues.

**Comment by powerapproach** [comment]
Hi Patrick,
Thanks for your reply.  There was nothing wrong with my executable - I just did not know how to drive it (starting with pressing the number keys).  Having done that the program works as intended.   I have found out the rest of the user interface from the code.

I have just one thing to say - amazing!  Other animations of horospheres, for example,  just give a glimpse of these objects.  Playing with some of the code (I have extended the horosphere plane further towards infinity - amazing again)

Interested to know whether it is possible to define ideal points in your code - want to try defining an ideal triangle, then attempt an ideal polyhedron.

Cheers
Adam Dickson

**Comment by patowen** [comment]
I'm glad you got it to work and that you were able to find and work with ModelBank. I'll re-title the issue to remind myself to write some documentation for how to use the program.

I checked, and it is possible to define ideal points. This engine uses the hyperboloid model of hyperbolic space, so points are denoted as (x, y, z, w) with w^2 - (x^2 + y^2 + z^2) = 1. If this equation does not hold, due to the way rendering works, x, y, z, and w are all effectively scaled evenly until the equation does hold. Because of this, to define a point at infinity, you need to make w^2 - (x^2 + y^2 + z^2) = 0. I haven't tested this extensively, but based on minimal testing, it seems to work well enough to try out.

One thing to note is that precision limitations of floating point arithmetic will become apparent relatively quickly as you approach one of these ideal points with the camera.

**Comment by patowen** [comment]
Should be fixed now. An extra step was added to the static build README.
